src/lammpsio/topology.py

```diff
@@ -241,3 +259,49 @@ class Impropers(Topology):
 
     def __init__(self, N, num_types=None):
         super().__init__(N=N, num_members=4, num_types=num_types)
+
+
+class LabelMap(MutableMapping):
+    def __init__(self, map=None):
+        self._map = {}
+        if map is not None:
+            self._map.update(map)
```

Comment:
I think you can just invoke `self.update` directly. That way you rely on the setters to do everything.
```suggestion
        if map is not None:
            self.update(map)
```